cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/main/MessageUtil.java

```diff
@@ -187,7 +187,15 @@ private static String getLatestMessageId(List<Location> locationsTimeSorted) thr
         DataFileReader<GenericRecord> dataFileReader =
           new DataFileReader<>(new File(latestLocation.toURI()),
                                new GenericDatumReader<>(ProgramRunInfoSerializer.SCHEMA));
-        //TODO - efficiently skip using sync points CDAP-13400
+        long skipLen = latestLocation.length() / 10;
+        long skipPoint = 0;
+        while (dataFileReader.hasNext()) {
```

Comment:
if `latestLocation.length() < 10`, will this be an infinite loop?

Reply:
yea, good point, however since we write schema even when there are no records, its always greater than 10 bytes, but its important to handle the case. fixed it now.